src/components/Editor/EditorMenu.js

```diff
@@ -78,7 +79,10 @@ function getMenuItems(
     label: revealInTreeLabel,
     accesskey: revealInTreeKey,
     disabled: false,
-    click: () => showSource(selectedSource.get("id"))
+    click: () => {
+      clearShownSource();
+      showSource(selectedSource.get("id"));
```

Comment:
i feel like this way, its clearer what is going on and each one function has its own responsibility. makes it easier to compose -- i vote for keeping it!